### PR TITLE
utils.py/get_key_def(): allow "expected_type=str" even with "to_path=True" param

### PR DESCRIPTION
**Pull request by remtav**
inference_segmentation.py: restore str as expected type to better reflect documentation

Documentation for "expected_type" in get_key_def(): [:param expected_type: (type) type of the expected variable.](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/utils.py#L142)

To me, this suggests that the input variable must bear this "expected_type". It feels confusing to expected a type that is actually one that is assigned during the get_key_def() function, "Path" in this case.
@CharlesAuthier is this slight retroactive change ok to you?

### CODE REVIEW

**Comment by CharlesAuthier** [comment]
I dont care but change the test too if you want to pass the check

**Comment by CharlesAuthier** [comment]
https://github.com/NRCan/geo-deep-learning/blob/develop/tests/utils/test_utils.py#L54

File: utils/utils.py [hunk 1]
```diff
@@ -175,6 +175,7 @@ def get_key_def(key, config, default=None, expected_type=None, to_path: bool = F
             val = Path(to_absolute_path(val))
         except TypeError:
             logging.error(f"Couldn't convert value {val} to a pathlib.Path object")
+        expected_type = Path if expected_type == str else expected_type  # allows "str" and "Path" as expected_type
```
Comment (CharlesAuthier) [inline utils/utils.py:178]: change your `expected_type == str` for ` isinstance(expected_type, str)` it's possible that python dont like when you use `==` to compare type

**Review by CharlesAuthier — CHANGES_REQUESTED** [review]

**Comment by CharlesAuthier** [comment]
btw can you add a test to check that pls

File: utils/utils.py (hunk 1 above)
Comment (remtav) [reply utils/utils.py:178]: Makes sense, but doesn't seem to be working for some reason:
image

**Review by ms888ekb — APPROVED** [review]

**Review by ms888ekb — APPROVED** [review]

**Review by valhassan — APPROVED** [review]

File: utils/utils.py (hunk 1 above)
Comment (remtav) [reply utils/utils.py:178]: Actually, I understood: isinstance() check the type of str. It's type is a Python "type" object, so the "expected_type" is itself a type class of type "str", not a type class. That's super hard to explain and would need a bit more details, but not too important at this point. :)

**Review by CharlesAuthier — APPROVED** [review]

**Pull request merged**